core/src/main/java/org/apache/calcite/rel/metadata/RelMetadataQuery.java

```diff
@@ -351,11 +353,32 @@ public Double getPercentageOriginalRows(RelNode rel) {
    * returns null
    */
   public RelColumnOrigin getColumnOrigin(RelNode rel, int column) {
+    return getColumnOrigin(rel, column, false);
+  }
+
+  /**
+   * Determines the origin of a column. If the param derived is false, provided the column maps
+   * to a single column that isn't derived, otherwise the column is derived.
```

Comment:
I think what you mean is 
"
Determines the origin of a column, allow users to choose whether to include the derived column."